x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/rule_upgrade/translations.tsx

```diff
@@ -0,0 +1,118 @@
+import { useKibana } from '../../../../../../common/lib/kibana/kibana_react';
+
+export const NUM_OF_FIELDS_WITH_UPDATES = (count: number) => (
+  <FormattedMessage
+    id="xpack.securitySolution.detectionEngine.rules.upgradeRules.diffTab.fieldsWithUpdates"
+    defaultMessage="Update has {count} {count, plural, one {field} other {fields}}"
+    values={{ count: <strong>{count}</strong> }}
+  />
+);
+
+export const NUM_OF_SOLVED_CONFLICTS = (count: number) => (
+  <FormattedMessage
+    id="xpack.securitySolution.detectionEngine.rules.upgradeRules.diffTab.numOfSolvableConflicts"
+    defaultMessage="{count} solved {count, plural, one {conflict} other {conflicts}}"
+    values={{ count: <strong>{count}</strong> }}
+  />
+);
+
+export const NUM_OF_UNSOLVED_CONFLICTS = (count: number) => (
+  <FormattedMessage
+    id="xpack.securitySolution.detectionEngine.rules.upgradeRules.diffTab.numOfNonSolvableConflicts"
+    defaultMessage="{count} unsolved {count, plural, one {conflict} other {conflicts}}"
+    values={{ count: <strong>{count}</strong> }}
+  />
+);
+
+const UPGRADE_RULES_DOCS_LINK = i18n.translate(
+  'xpack.securitySolution.detectionEngine.rules.upgradeRules.updateYourRulesDocsLink',
+  {
+    defaultMessage: 'update your rules',
+  }
+);
+
+export function RuleUpgradeHelper(): JSX.Element {
+  const {
+    docLinks: {
+      links: {
+        securitySolution: { manageDetectionRules },
+      },
+    },
+  } = useKibana().services;
+  const manageDetectionRulesSnoozeSection = `${manageDetectionRules}#edit-rules-settings`;
+
+  return (
+    <FormattedMessage
+      id="xpack.securitySolution.detectionEngine.rules.upgradeRules.ruleUpgradeHelper"
+      defaultMessage="Understand how to&nbsp;{docsLink}."
+      values={{
+        docsLink: (
+          <EuiLink href={manageDetectionRulesSnoozeSection} target="_blank">
+            {UPGRADE_RULES_DOCS_LINK}
+          </EuiLink>
+        ),
+      }}
+    />
+  );
+}
+
+export const UPGRADE_STATUS = i18n.translate(
+  'xpack.securitySolution.detectionEngine.rules.upgradeRules.fieldUpgradeState.upgradeStatusTitle',
+  {
+    defaultMessage: 'Update status:',
+  }
+);
+
+export const RULE_HAS_NON_SOLVABLE_CONFLICTS = (count: number) =>
+  i18n.translate(
+    'xpack.securitySolution.detectionEngine.rules.upgradeRules.fieldUpgradeState.ruleHasNonSolvableConflicts',
+    {
+      values: { count },
+      defaultMessage:
+        '{count} {count, plural, one {field has a unsolved conflict} other {fields have unsolved conflicts}}. Please review and provide a final update.',
+    }
+  );
+
+export const RULE_HAS_NON_SOLVABLE_CONFLICTS_DESCRIPTION = i18n.translate(
+  'xpack.securitySolution.detectionEngine.rules.upgradeRules.fieldUpgradeState.ruleHasNonSolvableConflictsDescription',
+  {
+    defaultMessage:
+      'Please provide an input for the unsolved conflict. You can also keep the current without the updates, or accept the Elastic update but lose your modifications.',
```

Comment:
nit:
```suggestion
      'Please provide an input for the unsolved conflict. You can also keep the current version without the updates, or accept the Elastic update but lose your modifications.',
```

I know this text came from the designs. We'll need to make sure docs folks proofread this.